src/System.Memory/src/System/Buffers/ReadOnlyBuffer_helpers.cs

```diff
@@ -0,0 +1,305 @@
+            endIndex = GetIndex(endIndex);
+
+            switch (type)
+            {
+                case BufferType.MemoryList:
+                    if (start.Segment == end.Segment && endIndex - startIndex >= count)
+                    {
+                        return new SequencePosition(start.Segment, startIndex + count);
+                    }
+                    return SeekMultiSegment((IMemoryList<byte>)start.Segment, startIndex, (IMemoryList<byte>)end.Segment, endIndex, count, checkEndReachable);
+
+                case BufferType.OwnedMemory:
+                case BufferType.Array:
+                    if (start.Segment != end.Segment)
+                    {
+                        ThrowHelper.ThrowInvalidOperationException_EndPositionNotReached();
+                    }
+
+                    if (endIndex - startIndex >= count)
+                    {
+                        return new SequencePosition(start.Segment, startIndex + count);
+                    }
+
+                    ThrowHelper.ThrowArgumentOutOfRangeException_CountOutOfRange();
+                    return default;
+
+                default:
+                    ThrowHelper.ThrowInvalidOperationException_UnexpectedSegmentType();
+                    return default;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal SequencePosition Seek(SequencePosition start, SequencePosition end, long count, bool checkEndReachable = true)
+        {
+            int startIndex = start.Index;
+            int endIndex = end.Index;
+            BufferType type = GetBufferType();
+
+            startIndex = GetIndex(startIndex);
+            endIndex = GetIndex(endIndex);
+
+            switch (type)
+            {
+                case BufferType.MemoryList:
+                    if (start.Segment == end.Segment && endIndex - startIndex >= count)
+                    {
+                        // end.Index >= count + Index and end.Index is int
+                        return new SequencePosition(start.Segment, startIndex + (int)count);
+                    }
+                    return SeekMultiSegment((IMemoryList<byte>)start.Segment, startIndex, (IMemoryList<byte>)end.Segment, endIndex, count, checkEndReachable);
+
+                case BufferType.OwnedMemory:
+                case BufferType.Array:
+                    if (endIndex - startIndex >= count)
+                    {
+                        // end.Index >= count + Index and end.Index is int
+                        return new SequencePosition(start.Segment, startIndex + (int)count);
+                    }
+
+                    ThrowHelper.ThrowArgumentOutOfRangeException_CountOutOfRange();
+                    return default;
+
+                default:
+                    ThrowHelper.ThrowInvalidOperationException_UnexpectedSegmentType();
+                    return default;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Why is this attribute necessary here?

Reply:
We'll review attributes as part of another work item.